client/signup/main.jsx

```diff
@@ -44,29 +47,30 @@ import { loadTrackingTool } from 'state/analytics/actions';
  */
 const MINIMUM_TIME_LOADING_SCREEN_IS_DISPLAYED = 3000;
 
-const Signup = React.createClass( {
-	displayName: 'Signup',
+class Signup extends React.Component {
+	static displayName = 'Signup';
 
-	contextTypes: {
-		store: React.PropTypes.object
-	},
+	static contextTypes = {
+		store: PropTypes.object
+	};
 
-	getInitialState() {
-		SignupDependencyStore.setReduxStore( this.context.store );
+	constructor( props, context ) {
+		super( props, context );
+		SignupDependencyStore.setReduxStore( context.store );
```

Comment:
Unrelated: can we refactor `SignupDependencyStore` to be a middleware and drop it from here?

Reply:
Patches welcome :)